src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -3128,4 +3130,54 @@ export default class EthereumApi implements Api {
     return "2";
   }
   //#endregion
+
+  //#region txpool
+
+  /**
+   * Returns the current content of the transaction pool.
+   *
+   * @returns The transactions currently pending or queued in the transaction pool.
+   * @example
+   * ```javascript
+   * const [from] = await provider.request({ method: "eth_accounts", params: [] });
+   * const pendingTx = await provider.request({ method: "eth_sendTransaction", params: [{ from, gas: "0x5b8d80", nonce:"0x0" }] });
+   * const queuedTx = await provider.request({ method: "eth_sendTransaction", params: [{ from, gas: "0x5b8d80", nonce:"0x2" }] });
+   * const pool = await provider.send("txpool_content");
+   * console.log(pool);
+   * ```
+   */
+  @assertArgLength(0)
+  async txpool_content(): Promise<{
+    pending: Map<string, Map<string, TypedTransactionJSON>>;
+    queued: Map<string, Map<string, TypedTransactionJSON>>;
+  }> {
+    const { transactions, common } = this.#blockchain;
+    const { transactionPool } = transactions;
+
+    const processMap = (map: Map<string, Heap<TypedTransaction>>) => {
+      let res = new Map<string, Map<string, TypedTransactionJSON>>();
+      for (let [_, transactions] of map) {
+        const arr = transactions.array;
+        for (let i = 0; i < transactions.length; ++i) {
+          const tx = arr[i];
+          const from = tx.from.toString();
+          if (res[from] === undefined) {
+            res[from] = {};
+          }
+          // The nonce keys are actual decimal numbers (as strings) and not
+          // hex literals (based on what geth returns).
+          const nonce = tx.nonce.toBigInt().toString();
+          res[from][nonce] = tx.toJSON(common);
+        }
+      }
+      return res;
```

Comment:
Can this be shifted to a separate utility? since other `txpool_*` methods might need it. If yes, which place is most suitable?